sdk/metrics/src/jmh/java/io/opentelemetry/sdk/metrics/internal/aggregator/HistogramBenchmark.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.aggregator;
+
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Measurement(iterations = 10, time = 1)
+@Warmup(iterations = 5, time = 1)
+@Fork(1)
+public class HistogramBenchmark {
+
+  @State(Scope.Thread)
+  public static class ThreadState {
+    @Param HistogramValueGenerator valueGen;
+    @Param HistogramAggregationParam aggregation;
+    private AggregatorHandle<?> aggregatorHandle;
+
+    @Setup(Level.Trial)
+    public final void setup() {
+      aggregatorHandle = aggregation.getAggregator().createHandle();
+    }
+
+    public void record() {
+      this.aggregatorHandle.recordDouble(valueGen.generateValue());
+    }
+  }
+
+  @Benchmark
```

Comment:
> number of threads appears to impact Exponential histogram more than ExplicitBucket, need to investigate if this is an artifact or real. 

I think this makes sense, since the exponential histogram has to spend more time in write lock, especially during rescaling. I am however kind of surprised that there isn't more variation/outliers across different recordings in the exponential histogram. 

I'd expect recordings that require rescaling to be much slower than recordings that fit into the existing range (I'm assuming the black horizontal lines indicate min/max time, correct me if I'm wrong). It appears that rescaling would be performed during warmup iterations which aren't measured. 

If we don't want to measure that outlier case of rescaling then that's reasonable, but it may also be interesting to see by how much a rescale slows down a recording.

Reply:
I think we need to measure both, and benchmarking here is hard.

1. I actually *want* us to try to optimise for stable scale per-metric-stream.   Ideally we find a way for the scale to stabilize for a particular metric so re-scaling *should* be rare / mostly seen at start-up.
2. We should understand the cost of rescaling and be able to identify whether we're doing it too often.  I'll look into adding a benchmark specific to that.   Are there other scenarios you can call out that should be measured?

Reply:
Great that sounds good. There aren't other scenarios I can think of currently